src/tests/ftest/util/general_utils.py

```diff
@@ -663,16 +668,34 @@ def stop_processes(hosts, pattern, verbose=True, timeout=60, added_filter=None):
     """
     result = {}
     log = getLogger()
-    log.info("Killing any processes on %s that match: %s", hosts, pattern)
+    if send_sigusr2 is True:
+        log.info("First dumping ULT stacks, then Killing any processes on %s "
+                 "that match: %s", hosts, pattern)
+    else:
+        log.info("Killing any processes on %s that match: %s", hosts, pattern)
 
     if added_filter:
-        ps_cmd = "/usr/bin/ps x | grep -E {} | grep -vE {}".format(
+        ps_cmd = "/usr/bin/ps xa | grep -E {} | grep -vE {}".format(
             pattern, added_filter)
     else:
         ps_cmd = "/usr/bin/pgrep --list-full {}".format(pattern)
 
     if hosts is not None:
-        commands = [
+        if send_sigusr2 is True:
+            commands_part1 = [
+                "rc=0",
+                "if " + ps_cmd,
+                "then rc=1",
+                "sudo pkill --signal USR2 {}".format(pattern),
+                # leave time for ABT info/stacks dump vs xstream/pool/ULT number
+		"sleep 20",
+                "fi",
+                "exit $rc",
+            ]
```

Comment:
Why not just add these to the existing `commands` list?

Reply:
I find this way more clear